packages/grid/_modules_/grid/GridComponent.tsx

```diff
@@ -81,7 +81,7 @@ export const GridComponent = React.forwardRef<HTMLDivElement, GridComponentProps
     useSorting(apiRef, props.rows);
     useColumnMenu(apiRef);
     usePreferencesPanel(apiRef);
-    useFilter(apiRef);
+    useFilter(apiRef, props.rows);
```

Comment:
Wouldn't it be better if you select the `rows` from the state and from within the hook? That way the hook will only depend on the `apiRef` to be passed as a param.

Reply:
it would be but, I need the prop to trigger the reset of filters as early as possible. If I use the rows then we will process the change, render, and useFilter will see the change and reapply the filters so we will have an extra rendering

Reply:
What if the rows are coming from another source? Not the props, like the apiRef?

```suggestion
    useFilter(apiRef);
```